README.md

```diff
@@ -1,28 +1,111 @@
 
 ## A Proposal
 
-Accepting the post-Spectre threat model means that we must begin work in earnest to unblock origin-level process isolation by removing the `document.domain` setter from the platform, sooner rather than later. Given its broad usage, an iterative approach to removal seems reasonable.
-
-We can begin with simple subsetting: the setter's effect on a origin's [domain](https://html.spec.whatwg.org/#concept-origin-domain) should be a no-op in the presence of `Cross-Origin-Opener-Policy` headers, as it already is for `Origin-Isolation` and `crossOriginIsolated` documents.
-
-Following that, we can create a ratcheting deprecation mechanism by introducing a `document-domain` Document Policy configuration point, with a default value of `false` (after an appropriate period of developer-facing messaging in places like browsers developer console, Lighthouse, etc). This policy would similarly disable `document.domain` by default, while still providing developers the ability to opt-into insecurity by sending an explicit `Document-Policy: document-domain` header along with their response. This explicit opt-in will allow us to track usage via HTTP Archive and browser telemetry, and to make appropriate decisions about a document's process when committing a navigation.
-
-As we reduce usage over time, we can strengthen this initial shift by requiring additional opt-in. Enterprise policies and reverse origin trials come to mind as subsequent milestones.
+The [`Òrigin-Agent-Cluster:` http header](`Origin-Agent-Cluster:` header](https://web.dev/origin-agent-cluster/)
+([spec](https://html.spec.whatwg.org/multipage/origin.html#origin-keyed-agent-clusters))
+allows a page to request being isolated by origin (instead of site). If set
+`true`(`Origin-Agent-Cluster: ?1`), the browser is asked to isolate pages by
+origin. (Agent Cluster is spec speak for isolation groups. Since the low-level
+isolation is not visible at the API layer, specifications only cursorily
+touches the subject.) As a side effect, writing to the  document.domain`
+accessor is blocked.
+
+Currently, bsence of the `Origin-Agent-Cluster:` header`defaults to `false`,
+meaning that an absent header forces isoaltion by site, rather than by origin,
+and setting `document.domain` continues to work. The proposal here is to change
+this default.
+
+In detail:
+
+* The `Origin-Agent-Cluster:` header will, when present, continue to work as
+  it currently does. What will change is the default when the header is absent.
+* We'll Implement a console warning when a page would be affected by this
+  change: If the page assigns to `document.domain` but does not have
+  any `Origin-Agent-Cluster:` header set a warning is issues.
+* We'll have a developer-enable-able feature flag that treats absence of
+  `Agent-Origin-Cluster:` header as having value true (`?1`)
+* Then we wait.
+* After developers have had some time to adjust, flip the flag, but leave it
+  disable-able.
```

Comment:
What does "leave it disable-able" mean?

I think it might be pointing toward enterprise policies? If so, mentioning that explicitly by pulling the last bullet up here might be clearer.

Reply:
What I meant is that the flag would remain, and that developers can now disable the feature in the same way they could earlier enable it. I clarified the wording. Although maybe I should just drop it, since it doesn't really add much to the plan.